external/llvm-project/mlir/lib/ExecutionEngine/CMakeLists.txt

```diff
@@ -280,4 +288,33 @@ if(MLIR_ENABLE_ROCM_RUNNER)
     PUBLIC
     ${ROCM_RUNTIME_LIBRARY}
   )
+
+  # Set HIP compile-time flags.
+  add_definitions(-D__HIP_PLATFORM_AMD__)
+
+  set_source_files_properties(RocmSystemDetect.cpp PROPERTIES
+    COMPILE_OPTIONS "-Wno-gnu-anonymous-struct;-Wno-nested-anon-types;-Wno-c++98-compat-extra-semi")
+  
+  add_mlir_library(MLIRRocmExecutionEngineUtils
+    RocmSystemDetect.cpp
+
+    ADDITIONAL_HEADER_DIRS
+    ${MLIR_MAIN_INCLUDE_DIR}/mlir/ExecutionEngine
+
+    DEPENDS
+    intrinsics_gen
+  )
+
+  llvm_update_compile_flags(MLIRRocmExecutionEngineUtils)
+  target_include_directories(MLIRRocmExecutionEngineUtils
+    PRIVATE
+    ${HIP_PATH}/include
```

Comment:
Repercussions because of this:
 1. You can't assume HIP must exist in the current environment, therefore this has to be guarded by the same macro from mlir ROCm runner
 2. Because of previous bullet point, you have to have a fall back for any component that rely on that this utils shared lib API
 3. Thinking of this, I don't believe it is easy enough to come up with the fall back that populate those fields in a meaningful way. I'm against of the idea to populate this field in a shared library. This should be a standalone executable that (1) If it functions, it gives the values for existing system (2) If now, user must put input in those field.

Reply:
Reminded by @krzysz00's comments, I think a better idea instead is to make a minimal executable for obtaining information from HIP, and make sure that executable run only once. Then the rest of test can skip calling HIP API again and again, which tends to cause trouble.

Reply:
This is guarded by the same macro (MLIR_ENABLE_ROCM_RUNNER), the same place mlir_rocm_runtime is built.

This lib (MLIRRocmExecutionEngineUtils) is only needed by xmir-runner which is also guarded.

Reply:
Gotcha! That's right somehow I missed the point that the newly added block is already wrapped in the macro. Sorry!